### PR TITLE
Support Python 3.10

### LINKED ISSUE #5146: Support Python 3.10
According to the Python 3.10 release schedule ([PEP 619](https://www.python.org/dev/peps/pep-0619/)), the second release candidate is scheduled to be released on 2021-09-07 with the final release on 2021-10-04. 

RC 2 is available for download: https://www.python.org/downloads/release/python-3100rc2/

I will track preparations to support Python 3.10 on this issue.

### PR DESCRIPTION
**Pull request by csadorf**
Closes #5146 .

### CODE REVIEW

**Comment by csadorf** [comment]
Status update:

- pip installation and tests pass
- conda installation still fails

**Comment by csadorf** [comment]
The issues with the conda environment are now reported here: https://github.com/s-weigand/setup-conda/issues/169

**Comment by csadorf** [comment]
Blocked by #5231 .

**Comment by sphuber** [comment]
This looks to be ready to go now, right @csadorf ? One question I had is whether we should change the `tests` job of the `ci-code.yml` workflow to run against Python 3.10 instead of 3.8. I think the original idea was that we run it against the oldest and newest supported version of Python. Currently it is 3.7 and 3.8 (we forgot to change this when we added 3.9) but maybe now it would be useful to change for 3.10?

**Comment by csadorf** [comment]
> This looks to be ready to go now, right @csadorf ? 

Yes, this is almost ready. There is an [issue with circus](https://github.com/aiidateam/aiida-core/issues/5146#issuecomment-973855900) that we should probably wait on to be resolved before merging this.

> One question I had is whether we should change the `tests` job of the `ci-code.yml` workflow to run against Python 3.10 instead of 3.8. I think the original idea was that we run it against the oldest and newest supported version of Python. Currently it is 3.7 and 3.8 (we forgot to change this when we added 3.9) but maybe now it would be useful to change for 3.10?

If I remember correctly, I don't think we forgot, the issue was that the pip installation was possible with 3.9, but still took significantly longer, because some dependencies needed to be compiled since no wheels where available. We might face the same issue here, but I am going to look into updating it.

File: requirements/requirements-py-3.10.txt
Comment (chrisjsewell) [inline requirements/requirements-py-3.10.txt:8]:
```suggestion
archive-path==0.3.6
```

File: requirements/requirements-py-3.10.txt
Comment (chrisjsewell) [inline requirements/requirements-py-3.10.txt:147]:
```suggestion
sphinx-panels==0.5.2
sphinx-sqlalchemy==0.1.1
```

**Review by chrisjsewell — DISMISSED** [review]
updates

**Comment by csadorf** [comment]
@chrisjsewell Should be ready to merge, I enabled auto-merge.

**Review by sphuber — APPROVED** [review]
Thanks @csadorf

**Comment by ltalirz** [comment]
@csadorf Just mentioning that I'm getting
```
ERROR: Could not find a version that satisfies the requirement spglib==1.16.2 (from versions: 1.6.0, 1.6.0.1, 1.6.0.2, 1.6.0.3, 1.9.0.5, 1.9.0.52, 1.9.0.53, 1.9.0.54, 1.9.2.13, 1.9.3, 1.9.3.1, 1.9.3.4, 1.9.4.2, 1.9.4.3, 1.9.4.4, 1.9.5, 1.9.6, 1.9.7, 1.9.7.1, 1.9.7.9, 1.9.7.17, 1.9.8, 1.9.8.7, 1.9.8.9, 1.9.8.11, 1.9.8.15, 1.9.8.20, 1.9.9, 1.9.9.5, 1.9.9.11, 1.9.9.15, 1.9.9.18, 1.9.9.25, 1.9.9.29, 1.9.9.38, 1.9.9.44, 1.9.9.116, 1.9.9.121, 1.9.10, 1.9.10.1, 1.10.0, 1.10.0.1, 1.10.0.2, 1.10.1, 1.10.1.4, 1.10.2, 1.10.2.4, 1.10.3, 1.10.3.5, 1.10.3.12, 1.10.3.14, 1.10.3.20, 1.10.3.23, 1.10.3.33, 1.10.3.39, 1.10.3.41, 1.10.3.43, 1.10.3.50, 1.10.3.52, 1.10.3.54, 1.10.3.56, 1.10.3.62, 1.10.3.65, 1.10.3.75, 1.10.4, 1.10.4.1, 1.10.4.7, 1.10.4.11, 1.11.0.post18, 1.11.0.post19, 1.11.1.post3, 1.11.2.post0, 1.11.2.post1, 1.12.0.post3, 1.12.0.post5, 1.12.1.post0, 1.12.1.post7, 1.12.2.post0, 1.13.0.post0, 1.13.0.post1, 1.13.0.post2, 1.13.0.post3, 1.13.0.post4, 1.13.0.post5, 1.13.0.post6, 1.14.0.post0, 1.14.0.post23, 1.14.1.post0, 1.15.0, 1.15.1, 1.16.0, 1.16.1)
```

This is on my macbook M1 + also on Chris's x86 macbook.
We cannot `pip install spglib==1.16.2`.

It's not quite clear to my why... perhaps there is no wheel uploaded for macos?

**Comment by ltalirz** [comment]
Opened https://github.com/spglib/spglib/issues/143

I'll open a PR on aiida-core to downgrade spglib

**Comment by csadorf** [comment]
Indeed, likely an architecture issue. Looks like we might have to start testing on different architectures to catch this... 🙄 Thanks for the PR!

**Pull request merged**
